src/main/api/studio-api.yaml

```diff
@@ -8981,21 +8091,49 @@ paths:
         '500':
           $ref: '#/components/responses/InternalServerError'
 
-  /api/2/workflow/packages/{id}:
-    get:
+  /api/2/workflow/{site}/package/{package}/approve:
+    post:
       tags:
         - workflow
-      summary: Get workflow package by id
-      description: Required permission "request_publish, publish"
-      operationId: workflowGetPackage
+      summary: Approve request publish submission
+      description: 'Required permission "publish"'
+      operationId: workflowApprove
       parameters:
-        - name: id
+        - name: site
           in: path
-          description: The workflow package ID
+          description: site ID
           required: true
           schema:
             type: string
-            format: UUID
+            example: "site123"
+        - name: package
+          in: path
+          description: package ID
+          required: true
+          schema:
+            type: integer
+            format: int64
+            example: "12345"
+      requestBody:
+        description: parameters for publish package approval
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                comment:
+                  type: string
+                  description: Reviewer's rejection comment
+                schedule:
+                  type: string
+                  format: datetime
+                  description: schedule when to publish content
+                updateSchedule:
+                    type: boolean
+                    description: if true, update schedule. Otherwise, keep the original schedule
```

Comment:
@sumerjabri @jmendeza Did we consider taking out the `updateSchedule` flag, and the schedule to be updated only if a new `schedule` is being submitted? What's the reason for the boolean together with the date?

Reply:
Because it could be null to remove schedule and publish right away
i.e.: It could be null either because it was not submitted or because null is meant to be the new value

Reply:
I see. Is there no difference between `null` or absent at the back? 
Alternatively, since the schedule is a string, we could also consider a constant "now" to indicate publish now instead of null, vs the date string (e.g. "2024-11-26T17:28:02.308Z") to set a new schedule date.